src/Symfony/Component/VarDumper/Caster/FFICaster.php

```diff
@@ -115,12 +115,20 @@ private static function castFFIPointer(Stub $stub, CType $type, ?CData $data = n
     private static function castFFIStringValue(CData $data): string|CutStub
     {
         $result = [];
+        $ffi = \FFI::cdef(<<<C
+            bool is_zend_mm(void);
+            bool is_zend_ptr(const void *ptr);
+        C);
 
         for ($i = 0; $i < self::MAX_STRING_LENGTH; ++$i) {
-            $result[$i] = $data[$i];
+            if ($ffi->is_zend_ptr($data) && $ffi->is_zend_mm()) {
+                $result[$i] = $data[$i];
 
-            if ("\0" === $result[$i]) {
-                return implode('', $result);
+                if ("\0" === $data[$i]) {
+                    return implode('', $result);
+                }
+            } else {
+                break;
```

Comment:
While testing I found that `is_zend_ptr()` is broken and may crash in some conditions. So my suggestion of using `is_zend_ptr()` is not good, sorry.

A safe alternative is to ensure that you only access memory in the same page as $data[0]: (it's safe as long as accessing $data[0] is safe)

``` php
        $ffi = \FFI::cdef(<<<C
            size_t zend_get_page_size(void);
        C);

        $pagesize = $ffi->zend_get_page_size();

        /* Get the address of $data */
        $start = $ffi->cast("uintptr_t", $ffi->cast("char*", $data))->cdata;
        /* Accessing memory in the same page as $start is safe */
        $max = min(self::MAX_STRING_LENGTH, ($start | ($pagesize - 1)) - $start);

        for ($i = 0; $i < $max; ++$i) {
            $result[$i] = $data[$i];

            if ("\0" === $data[$i]) {
                return implode('', $result);
            }
        }
```

Reply:
I couldn't find a proper way to get the page size. This looks good, thanks Arnaud!